### DISCUSSION TITLE
How can I visualize a subgraph of my heterogenous graph?

### DISCUSSION THREAD

**Discussion by arymandeshwal** [pyg-team/pytorch_geometric#7652] [Q&A] [answered]
So I created a link prediction model and explained it via CaptumExplainer and Explainer. After I get the subgraph from 
`explanation.get_complement_subgraph()` I would like to visualize it, however for Heterogenous graphs, there's no method to visualize it yet. 
And can anyone please explain some of my other doubts regarding Explainer, I am relatively new to it.

**Comment by rusty1s** [comment]
I am sorry for that. What you can try is to convert the `HeteroExplanation` object to an `Explanation` object by calling `to_homogeneous`, and visualize from there. This should hopefully work :)

**Comment by arymandeshwal** [comment]
I tried `explanation.get_explanation_subgraph().to_homogeneous()` however, I am unable to use `visualize_graph()`

**Reply by rusty1s** [reply]
How come? Because it is no longer a `Explanation` object? What happens if you create the `Explanation` object from these attributes?
```python
data = Explanation.from_dict(explanation.to_homogeneous().to_dict())
```

**Reply by arymandeshwal** [reply]
is there any way to visualize it without converting it to homogenous as it loses information about node types.

**Comment by arymandeshwal** [comment]
Now it works, thanks a lot. However, is there any way to visualize it without converting it to homogenous as it loses information about node types.

**Reply by rusty1s** [reply]
Not yet, we track progress in this issue: https://github.com/pyg-team/pytorch_geometric/issues/7340

**Comment by rusty1s** [comment, marked as answer]
Added this via https://github.com/pyg-team/pytorch_geometric/pull/7713.
